### DISCUSSION TITLE
how to disable schema validation for a given file or file name pattern

### DISCUSSION THREAD

**Discussion by sgandon** [redhat-developer/vscode-yaml#962] [Q&A]
Hello,
We have an internal file called _component.yaml_ and this is validated with a "pipeline component" schema.
How can I disable the schema validation for this file or for all file with this name ?
I have added a comment at the top of the file like this
```
# yaml-language-server: $schema=<urlToTheSchema> this line removes the default schema used by vscode but idealy there should be our schema here.
```
this creates an error so no validation happens but this is a poor workaround.
If you have a more elegant way, that would be great !

**Comment by sbliven** [comment]
I wondered this as well.

You can disable validation for all files with `"yaml.validate": false`. However I wasn't able to disable it for a single file.

One work-around could be to make a very permissive schema that would accept all additional properties, and then choose this for the file from the bottom bar.

*Edit*: According to #245 the most permissive schema is an empty `{}`. One user published a file with that text as a gist, allowing validation to be disabled for a file like so:
```
# yaml-language-server: $schema=https://gist.githubusercontent.com/sargunv/c2ca41a08391cd06feaad97aece309e4/raw/empty-json-schema.json
```

**Comment by gilex-dev** [comment]
It seems that setting it to `none` works:
```yaml
# yaml-language-server: $schema=none
# rest of your yaml file...
```
Version 1.24.0 redhat.vscode-yaml with vscodium Version: 1.135.06055
